sql/moz-fx-data-shared-prod/firefox_ios/app_store_funnel_web_referrer/view.sql

```diff
@@ -0,0 +1,182 @@
+-- Each CTE is broken down into two sections that are unioned
+-- as we know only mozilla.org web_referrer has campaigns linked to it.
+-- This allows us to combine the campaign and web_referrer datasets together.
+CREATE MATERIALIZED VIEW `moz-fx-data-shared-prod.firefox_ios.app_store_funnel_web_referrer`
+PARTITION BY
+  `date`
+CLUSTER BY
+  app_id,
+  territory,
+  web_referrer
+OPTIONS
+  (
+    enable_refresh = TRUE,
+    refresh_interval_minutes = 720,
+    allow_non_incremental_definition = true,
+    max_staleness = INTERVAL 12 HOUR
+  )
+AS
+WITH views_territory AS (
+  SELECT
+    DATE(web_referrer.`date`) AS `date`,
+    web_referrer.app_id,
+    web_referrer.territory,
+    web_referrer.web_referrer,
+    COALESCE(campaign.campaign, "MISSING") AS campaign,
+    SUM(campaign.impressions) AS impressions,
+    SUM(campaign.impressions_unique_device) AS impressions_unique_device,
+    SUM(campaign.page_views) AS page_views,
+    SUM(campaign.page_views_unique_device) AS page_views_unique_device,
+  FROM
+    `moz-fx-data-bq-fivetran.firefox_app_store.app_store_territory_web_referrer_report` AS web_referrer
+  LEFT JOIN
+    `moz-fx-data-bq-fivetran.firefox_app_store.app_store_territory_campaign_report` AS campaign
+    USING (`date`, app_id, territory)
```

Comment:
Consider consolidating the `UNION ALL` into a single query statement, and limiting to the correct join rows using an `ON` clause